src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -69,6 +79,96 @@ private String setKey() {
     }
   }
 
+  /**
+   * Formats a date from YYYYMMDD format to YYYY-MM-DD format. This is to accommodate the historical
+   * price repository which automatically does this conversion.
+   * @param date A string representing a date in YYYYMMDD format.
+   * @return A string representing the same date in YYYY-MM-DD format.
+   */
+
+  private String formatDate(final String date) {
+    return date.substring(0, 4) + "-" + date.substring(4, 6) + "-" + date.substring(6);
```

Comment:
What if the date is null? What if the date is less than 6 characters? What if it's formatted differently?

Reply:
Manually date parsing is never a good idea... my general recommendation would be to use Date objects as much as possible and only parse/toString when needed.

Look into ```LocalDate.parse(...)```